.github/workflows/build.yml

```diff
@@ -0,0 +1,48 @@
+name: build
+
+on:
+  push:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install required packages
+        run: |
+          sudo apt install --no-install-recommends -y git cmake ninja-build gperf \
+            ccache dfu-util device-tree-compiler wget \
+            python3-dev python3-pip python3-setuptools python3-tk python3-wheel xz-utils file \
+            make gcc gcc-multilib g++-multilib libsdl2-dev libmagic1 python3-venv
+
+      - name: Install Zephyr
+        run: |
+          python3 -m venv $GITHUB_WORKSPACE/zephyrproject/.venv
+          source $GITHUB_WORKSPACE/zephyrproject/.venv/bin/activate
+          pip install west
+          west init -m https://github.com/eurus-project/zephyr $GITHUB_WORKSPACE/zephyrproject/ > /dev/null 2>&1
+          cd $GITHUB_WORKSPACE/zephyrproject/
+          west update
+          west zephyr-export
+          west packages pip --install
+
+      - name: Install Zephyr SDK
+        run: |
+          source $GITHUB_WORKSPACE/zephyrproject/.venv/bin/activate
+          cd $GITHUB_WORKSPACE/zephyrproject/
+          west sdk install > /dev/null 2>&1
+
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          path: zephyrproject/${{ github.repository }}
+      
+      - name: Build project
+        run: |
+          source $GITHUB_WORKSPACE/zephyrproject/.venv/bin/activate
+          cd $GITHUB_WORKSPACE/zephyrproject/${{ github.repository }}
+          west update
+          python3 -m venv python_venv
+          source python_venv/bin/activate
+          pip install -r requirements.txt
+          source $GITHUB_WORKSPACE/zephyrproject/.venv/bin/activate
```

Comment:
We don't actually need to activate the environment (and it might also conflict with zephyrs venv as well). Things which require our venv will call its python directly. We might want to reevaluate this approach in the future, but for now it's good enough.